pkg/3scale/amp/operator/system_reconciler.go

```diff
@@ -246,24 +246,35 @@ func (r *SystemReconciler) validateS3StorageProvidedConfiguration() error {
 
 	secretData := awsSecret.Data
 	var result *string
-	result = helper.GetSecretDataValue(secretData, component.AwsAccessKeyID)
-	if result == nil {
-		return fmt.Errorf("Secret field '%s' is required in secret '%s'", component.AwsAccessKeyID, awsCredentialsSecretName)
-	}
-
-	result = helper.GetSecretDataValue(secretData, component.AwsSecretAccessKey)
-	if result == nil {
-		return fmt.Errorf("Secret field '%s' is required in secret '%s'", component.AwsSecretAccessKey, awsCredentialsSecretName)
+	if system.Options.S3FileStorageOptions.S3STSEnabled {
+		result = helper.GetSecretDataValue(secretData, component.AwsRoleArn)
+		if result == nil {
+			return fmt.Errorf("secret field '%s' is required in secret '%s'", component.AwsRoleArn, awsCredentialsSecretName)
+		}
+		tokenPath := helper.GetSecretDataValue(secretData, component.AwsWebIdentityTokenFile)
+		if tokenPath == nil {
+			return fmt.Errorf("secret field '%s' is required in secret '%s'", component.AwsWebIdentityTokenFile, awsCredentialsSecretName)
+		}
+		system.Options.S3FileStorageOptions.TokenPath = *tokenPath
```

Comment:
I do not like this here. This method is for verification, it should not fill options. system_option_provider should do that. 

But it is good enough for now. I will open another PR when this is imerged to fix this